src/build-scripts/build_materialX_osl.py

```diff
@@ -186,21 +186,29 @@ def mx_to_osl(shader, shader_types, options):
     mx_code = open_mx_file(shader, options)
     if mx_code is not None:
         for var_type in shader_types:
-            osl_shadername = '%s_%s' % (shader, TYPE_STRING[var_type])            
-            if options['v']:
-                print('Building %s' % osl_shadername)
-            osl_code = mx_code.replace('SHADER_NAME(%s)' % shader, osl_shadername)
-            osl_code = osl_code.replace('#include \"mx_types.h\"', '#define %s 1\n#include \"mx_types.h\"' % var_type)
-            osl_code = re.sub(r'\bTYPE\b', SHADER_TYPES[var_type], osl_code)
-            osl_filepath = write_osl_file(osl_shadername, osl_code, options)
-            
-            # build oso bytecode if compile flag is on
-            if options['compile']:
-                oso_filename = '%s.oso'%(osl_shadername)
-                osl_filepath = '%s.osl' % (osl_shadername)
+            if var_type in SHADER_TYPES:
```

Comment:
might be more clear to rename shader_types or SHADER_TYPES so you don't have two variables with the same name.  
shader_types --> types_to_build